py/src/braintrust/logger.py

```diff
@@ -368,14 +392,24 @@ def flush(self, batch_size=100):
         # We cannot have multiple threads flushing in parallel, because the
         # order of published elements would be undefined.
         with self.flush_lock:
+            # Drain the queue.
+            all_items = []
+            try:
+                while True:
+                    all_items.append(self.queue.get_nowait())
```

Comment:
this is a scary piece of code. i assume we rely on the fact that the function will either return something or throw an exception. i'd still feel more comfortable if we capped/batched it somehow, e.g. to `num_threads*batch_size`?

Reply:
Why don't I cap it to `self.queue.qsize()` called at the beginning. That way we don't get into some weird game of pulling things off the queue on one thread and pushing them on another forever.

Reply:
👍